drivers/rtc/rtc_rpi_pico.c

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (c) 2023 Andrew Featherstobne
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/rtc.h>
+#include <zephyr/irq.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/spinlock.h>
+
+#include <hardware/irq.h>
+#include <hardware/rtc.h>
+#include <hardware/regs/rtc.h>
+
+#define DT_DRV_COMPAT raspberrypi_pico_rtc
+
+/* struct tm start time:   1st, Jan, 1900 */
+#define TM_YEAR_REF 1900
+static int rtc_rpi_pico_alarm_get_time(const struct device *dev, uint16_t id, uint16_t *mask,
+				       struct rtc_time *timeptr);
+struct rtc_rpi_pico_data {
+	struct k_spinlock lock;
```

Comment:
Any particular reason for using a spinlock and not a mutex? I see the stm32 driver uses a mutex, so it might be possible here too.

Reply:
The feedback I received previously at https://github.com/zephyrproject-rtos/zephyr/pull/64939#discussion_r1386162310 was to use a spinlock and not mutex. Can you add more detail on why you think this should be the other way around?